app/uk/gov/hmrc/economiccrimelevyregistration/connectors/IntegrationFrameworkConnector.scala

```diff
@@ -16,45 +16,73 @@
+  override val actorSystem: ActorSystem
+)(implicit ec: ExecutionContext)
+    extends BaseConnector
+    with Retries {
 
-  private def integrationFrameworkHeaders(bearerToken: String): Seq[(String, String)] = Seq(
-    (HeaderNames.AUTHORIZATION, s"Bearer $bearerToken"),
+  private def integrationFrameworkHeaders(correlationId: String): Seq[(String, String)] = Seq(
+    (HeaderNames.AUTHORIZATION, s"Bearer ${appConfig.getSubscriptionStatusBearerToken}"),
     (CustomHeaderNames.Environment, appConfig.integrationFrameworkEnvironment),
-    (CustomHeaderNames.CorrelationId, correlationIdGenerator.generateCorrelationId)
+    (CustomHeaderNames.CorrelationId, correlationId),
+    (HEADER_X_CORRELATION_ID, correlationId)
   )
 
   def getSubscriptionStatus(
     businessPartnerId: String
-  )(implicit hc: HeaderCarrier): Future[SubscriptionStatusResponse] =
-    httpClient.GET[SubscriptionStatusResponse](
-      s"${appConfig.integrationFrameworkUrl}/cross-regime/subscription/ECL/SAFE/$businessPartnerId/status",
-      headers = integrationFrameworkHeaders(appConfig.getSubscriptionStatusBearerToken)
-    )
+  )(implicit hc: HeaderCarrier): Future[SubscriptionStatusResponse] = {
+    val correlationId = createCorrelationId(hc)
+
+    retryFor[SubscriptionStatusResponse]("Get subscription status")(retryCondition) {
+      httpClient
+        .get(url"${appConfig.integrationFrameworkUrl}/cross-regime/subscription/ECL/SAFE/$businessPartnerId/status")
+        .setHeader(integrationFrameworkHeaders(correlationId): _*)
+        .executeAndDeserialise[SubscriptionStatusResponse]
+    }
+  }
 
   def subscribeToEcl(eclSubscription: EclSubscription)(implicit
     hc: HeaderCarrier
-  ): Future[Either[UpstreamErrorResponse, CreateEclSubscriptionResponse]] =
-    httpClient.POST[Subscription, Either[UpstreamErrorResponse, CreateEclSubscriptionResponse]](
-      s"${appConfig.integrationFrameworkUrl}/economic-crime-levy/subscription/${eclSubscription.businessPartnerId}",
-      eclSubscription.subscription,
-      headers = integrationFrameworkHeaders(appConfig.integrationFrameworkBearerToken)
-    )
+  ): Future[CreateEclSubscriptionResponse] = {
+    val correlationId: String = createCorrelationId(hc)
+
+    retryFor[CreateEclSubscriptionResponse]("Subscribe to ECL")(retryCondition) {
+      httpClient
+        .post(
+          url"${appConfig.integrationFrameworkUrl}/economic-crime-levy/subscription/${eclSubscription.businessPartnerId}"
+        )
+        .withBody(Json.toJson(eclSubscription))
+        .setHeader(integrationFrameworkHeaders(correlationId): _*)
+        .executeAndDeserialise[CreateEclSubscriptionResponse]
+    }
+  }
 
+  private def createCorrelationId(hc: HeaderCarrier) = {
+    val correlationId = hc.headers(Seq(CorrelationIdHelper.HEADER_X_CORRELATION_ID)) match {
+      case Nil                     => UUID.randomUUID().toString
+      case Seq((_, correlationId)) => correlationId
+    }
+    correlationId
```

Comment:
you don't need to save it in a val and return the val. The below is sufficient

```
private def createCorrelationId(hc: HeaderCarrier) = {
   hc.headers(Seq(CorrelationIdHelper.HEADER_X_CORRELATION_ID)) match {
       case Nil                     => UUID.randomUUID().toString
       case Seq((_, correlationId)) => correlationId
     }
}
```

Reply:
Ah, good catch, thank you :). Fixed